python/src/aiconfig/AIConfigSettings.py

```diff
@@ -384,7 +439,7 @@ def update_model(
                 )
             prompt.metadata.model = model_metadata
         else:
-            self.metadata.models[model_metadata.name] = model_metadata
+            self.metadata.models[model_metadata.name] = model_metadata.settings
```

Comment:
what if `.settings` doesn't exist?

Reply:
the logic flow ensures that .settings always exists; if input is of type dict, it gets converted into a ModelMetadata object which always has a `.settings` (defaulted to an empty dictionary)